### PR TITLE
refactor(recipes): add h100 recipe namespace

### PR DESCRIPTION
**Pull request by yaoyu-33**
## Summary
- Move library recipe implementations under hardware-specific h100 modules while keeping legacy recipe names as compatibility wrappers.
- Normalize canonical library PEFT recipe names to use `_peft_` instead of `_lora_`.
- Flatten Qwen3/Qwen3.5-VL recipe bodies and remove parameterized recipe helper patterns.
- Update recipe docs, Fern nightly pages, and recipe tests for the new naming.

## Split
This PR is now recipes-only. The legacy performance config removal has been split into #4623.

## Validation
- `python3 -m compileall -q src/megatron/bridge/recipes tests/unit_tests/recipes tests/functional_tests/test_groups/recipes/test_qwen35_vl_recipes_pretrain.py scripts/translate_mlm_to_bridge.py`
- `uvx pre-commit run --all-files`

Note: `uv run pre-commit run --all-files` is blocked on macOS by `nvidia-resiliency-ext==0.6.0` Linux-only wheels.

### CODE REVIEW

File: docs/fern/versions/nightly/pages/models/nemotron/nemotron-nano-v2-vl.mdx [hunk 1]
```diff
@@ -104,33 +104,13 @@ Note:
 
 ### Parameter-Efficient Finetuning (PEFT)
 Parameter-efficient finetuning (PEFT) using LoRA is supported.
-LoRA can be independently applied to the vision model, vision projection, and language model. We support two commonly used
```
Comment (yaoyu-33) [inline docs/fern/versions/nightly/pages/models/nemotron/nemotron-nano-v2-vl.mdx:107]: do not change the text here, not necessary.

File: docs/fern/versions/nightly/pages/models/nemotron/nemotron-nano-v2-vl.mdx [hunk 2]
```diff
@@ -104,33 +104,13 @@ Note:
...
-settings out of the box in the example script:
-1. Apply LoRA to the language model, and fully finetune the vision model and projection (used when the visual
-   distribution is substantially different from pretrained.)
+The PEFT recipe applies adapters to the default VLM target modules.
 
 ```bash
 uv run python -m torch.distributed.run --nproc-per-node=8 examples/models/nemotron/nemotron_vl/finetune_nemotron_nano_v2_vl.py \
 --hf-model-path $HF_MODEL_PATH \
 --pretrained-checkpoint $MEGATRON_MODEL_PATH \
---lora-on-language-model \
```
Comment (yaoyu-33) [inline docs/fern/versions/nightly/pages/models/nemotron/nemotron-nano-v2-vl.mdx:116]: do not remove this part, should still do 2 paths.

File: docs/fern/versions/nightly/pages/models/nemotron/nemotron-nano-v2-vl.mdx (hunk 1 above)
Comment (yaoyu-33) [inline docs/fern/versions/nightly/pages/models/nemotron/nemotron-nano-v2-vl.mdx:107]: similar as above. do not change text, do not remove one path for script

File: docs/fern/versions/nightly/pages/index.mdx [hunk 3]
```diff
@@ -132,14 +132,14 @@ Training quickstart using pre-configured recipes:
 
 ```python
 from megatron.bridge import AutoBridge
-from megatron.bridge.recipes.llama import llama32_1b_pretrain_config
+from megatron.bridge.recipes.llama.h100 import llama32_1b_pretrain_1gpu_h100_bf16_config
```
Comment (yaoyu-33) [inline docs/fern/versions/nightly/pages/index.mdx]: can we still import from `from megatron.bridge.recipes.llama`

File: docs/fern/versions/nightly/pages/recipe-usage.mdx [hunk 4]
```diff
@@ -12,7 +12,7 @@ This guide will cover the next steps to make use of a training recipe, including
 
 ## Choosing a recipe or a new config
 
-Start from an exported recipe when the model family and workflow already exist in `megatron.bridge.recipes`. Recipe functions such as `llama3_8b_pretrain_config`, `llama32_1b_sft_config`, and `qwen3_8b_peft_config` provide model, optimizer, scheduler, precision, dataset, logger, and checkpoint defaults in one `ConfigContainer`. Override those defaults for your dataset, checkpoint paths, run length, parallelism, or precision before creating a new recipe.
+Start from an exported recipe when the model family and workflow already exist in `megatron.bridge.recipes`. Hardware-specific H100 recipes such as `llama3_8b_pretrain_2gpu_h100_bf16_config`, `llama32_1b_sft_1gpu_h100_bf16_config`, and `qwen3_8b_peft_1gpu_h100_bf16_config` are the canonical recipe names and share the same `<model>_<task>_<num>gpu_<gpu>_<dtype>_config` naming strategy as performance recipes. The older recipe names remain available as compatibility wrappers. Override those defaults for your dataset, checkpoint paths, run length, parallelism, or precision before creating a new recipe.
```
Comment (yaoyu-33) [inline docs/fern/versions/nightly/pages/recipe-usage.mdx]: I think we have variant in the end? <model>_<task>_<num>gpu_<gpu>_<dtype>_<variant(optional)>_config - this should go into skills as well.

File: docs/models/nemotron/nemotron-nano-v2-vl.md [hunk 5]
```diff
@@ -104,33 +104,13 @@ Note:
 
 ### Parameter-Efficient Finetuning (PEFT)
 Parameter-efficient finetuning (PEFT) using LoRA is supported.
-LoRA can be independently applied to the vision model, vision projection, and language model. We support two commonly used
```
Comment (yaoyu-33) [inline docs/models/nemotron/nemotron-nano-v2-vl.md:107]: similar as before, these text should not change? command paths should not change either.

File: docs/recipe-usage.md [hunk 6]
```diff
@@ -12,7 +12,7 @@ This guide will cover the next steps to make use of a training recipe, including
 
 ## Choosing a recipe or a new config
 
-Start from an exported recipe when the model family and workflow already exist in `megatron.bridge.recipes`. Recipe functions such as `llama3_8b_pretrain_config`, `llama32_1b_sft_config`, and `qwen3_8b_peft_config` provide model, optimizer, scheduler, precision, dataset, logger, and checkpoint defaults in one `ConfigContainer`. Override those defaults for your dataset, checkpoint paths, run length, parallelism, or precision before creating a new recipe.
+Start from an exported recipe when the model family and workflow already exist in `megatron.bridge.recipes`. Hardware-specific H100 recipes such as `llama3_8b_pretrain_2gpu_h100_bf16_config`, `llama32_1b_sft_1gpu_h100_bf16_config`, and `qwen3_8b_peft_1gpu_h100_bf16_config` are the canonical recipe names and share the same `<model>_<task>_<num>gpu_<gpu>_<dtype>_config` naming strategy as performance recipes. The older recipe names remain available as compatibility wrappers. Override those defaults for your dataset, checkpoint paths, run length, parallelism, or precision before creating a new recipe.
```
Comment (yaoyu-33) [inline docs/recipe-usage.md]: similar as previous comments that needs to have variant in the canonical.

File: examples/models/nemotron/nemotron_vl/finetune_nemotron_nano_v2_vl.py [hunk 7]
```diff
@@ -80,14 +80,9 @@ def parse_cli_args() -> Tuple[argparse.Namespace, list[str]]:
         help="Path to a Megatron-Bridge checkpoint directory or HuggingFace model name to load weights from.",
     )
     parser.add_argument(
-        "--lora-on-language-model",
```
Comment (yaoyu-33) [inline examples/models/nemotron/nemotron_vl/finetune_nemotron_nano_v2_vl.py:83]: why remove one of the options? should still need to do 2 options? should not collapse.

File: examples/models/nemotron/nemotron_vl/finetune_nemotron_nano_v2_vl.py [hunk 8]
```diff
@@ -80,14 +80,9 @@ def parse_cli_args() -> Tuple[argparse.Namespace, list[str]]:
...
+        "--peft",
```
Comment (yaoyu-33) [inline examples/models/nemotron/nemotron_vl/finetune_nemotron_nano_v2_vl.py]: similar, should keep both `peft-on-language-model` etc.

File: examples/models/qwen/qwen_vl/finetune_qwen_vl.py [hunk 9]
```diff
@@ -44,27 +44,27 @@
     Finetune using the imported checkpoint:
         Qwen2.5-VL 3B:
             $ uv run python -m torch.distributed.run --nproc_per_node=8 examples/models/qwen/qwen_vl/finetune_qwen_vl.py \\
-                --recipe qwen25_vl_3b_finetune_config \\
+                --recipe qwen25_vl_3b_sft_config \\
```
Comment (yaoyu-33) [inline examples/models/qwen/qwen_vl/finetune_qwen_vl.py]: use the new name ; not the legacy name

**Comment by yaoyu-33** [comment]
/ok to test [c9473a2](https://github.com/NVIDIA-NeMo/Megatron-Bridge/pull/4622/commits/c9473a239db3ca5674e1ed9d8cf687fe612358b9)

File: src/megatron/bridge/recipes/utils/naming.py [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+from __future__ import annotations
+
+import re
+
+
+PRECISION_NAME_MAP = {
+    "bf16": "bf16",
+    "fp8_cs": "fp8cs",
+    "fp8_mx": "fp8mx",
+    "fp8_sc": "fp8sc",
```
Comment (cuichenx) [inline src/megatron/bridge/recipes/utils/naming.py:26]: worth adding a comment what cs mx sc are?

File: scripts/training/run_recipe.py
Comment (cuichenx) [inline scripts/training/run_recipe.py:302]: thanks for cleaning up the other args, i think this part still needs to be cleaned up. depending on what args.dataset is, args.packed_sequence etc may or may not take effect which is confusing

File: docs/recipe-usage.md [hunk 11]
```diff
@@ -12,7 +12,7 @@ This guide will cover the next steps to make use of a training recipe, including
 
 ## Choosing a recipe or a new config
 
-Start from an exported recipe when the model family and workflow already exist in `megatron.bridge.recipes`. Recipe functions such as `llama3_8b_pretrain_config`, `llama32_1b_sft_config`, and `qwen3_8b_peft_config` provide model, optimizer, scheduler, precision, dataset, logger, and checkpoint defaults in one `ConfigContainer`. Override those defaults for your dataset, checkpoint paths, run length, parallelism, or precision before creating a new recipe.
+Start from an exported recipe when the model family and workflow already exist in `megatron.bridge.recipes`. Hardware-specific H100 recipes such as `llama3_8b_pretrain_2gpu_h100_bf16_config`, `llama32_1b_sft_1gpu_h100_bf16_config`, and `qwen3_8b_peft_1gpu_h100_bf16_config` are the canonical recipe names and share the same `<model>_<task>_<num>gpu_<gpu>_<dtype>[_<variant>]_config` naming strategy as performance recipes. The older recipe names remain available as compatibility aliases. Override those defaults for your dataset, checkpoint paths, run length, parallelism, or precision before creating a new recipe.
```
Comment (cuichenx) [inline docs/recipe-usage.md]: mention other hardwares will come soon?

File: examples/models/qwen/qwen_vl/finetune_qwen_vl.py
Comment (cuichenx) [inline examples/models/qwen/qwen_vl/finetune_qwen_vl.py:1]: we should stop maintaining this script?

File: src/megatron/bridge/recipes/utils/naming.py [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+    config_variant: str | None = None,
+) -> str:
+    """Build a flat recipe function name from CLI dimensions."""
+    return (
+        f"{model_recipe_name}_{task}_{num_gpus}gpu_{gpu}_{normalize_precision_name(precision)}"
+        f"{recipe_variant_suffix(config_variant)}_config"
+    )
+
+
+def is_hardware_recipe_name(recipe_name: str) -> bool:
```
Comment (cuichenx) [inline src/megatron/bridge/recipes/utils/naming.py]: this is not used?

File: src/megatron/bridge/recipes/__init__.py [hunk 13]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+# Copyright (c) 2026, NVIDIA CORPORATION.  All rights reserved.
```
Comment (cuichenx) [inline src/megatron/bridge/recipes/__init__.py]: probably shouldn't need to change these copyright years?

**Review by cuichenx — COMMENTED** [review]
these also need to be updated for broken reference:
- tutorials/moe/moe.ipynb
- tutorials/training/reduced_precision_training.ipynb
- scripts/performance/README.md

File: examples/models/qwen/qwen_vl/finetune_qwen_vl.py
Comment (yaoyu-33) [reply examples/models/qwen/qwen_vl/finetune_qwen_vl.py:1]: yes, right should be removed.

File: scripts/training/run_recipe.py
Comment (yaoyu-33) [reply scripts/training/run_recipe.py:302]: yeah future pr.

**Comment by yaoyu-33** [comment]
/ok to test [e1edbd3](https://github.com/NVIDIA-NeMo/Megatron-Bridge/pull/4622/commits/e1edbd39d19fb1ba70fe915c303298308bd7e494)
